clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -1972,6 +1973,19 @@ ParseResult cir::FuncOp::parse(OpAsmParser &parser, OperationState &state) {
     state.addAttribute(visNameAttr,
                        parser.getBuilder().getStringAttr(visAttrStr));
   }
+
+  ::llvm::StringRef cirVisAttrStr;
+  if (parser.parseOptionalKeyword(&cirVisAttrStr, {"hidden"}).succeeded()) {
+    state.addAttribute(
+        visibilityNameAttr,
+        CIR_VisibilityAttr::get(parser.getContext(), VisibilityKind::Hidden));
+  } else if (parser.parseOptionalKeyword(&cirVisAttrStr, {"protected"})
+                 .succeeded()) {
+    state.addAttribute(visibilityNameAttr,
+                       CIR_VisibilityAttr::get(parser.getContext(),
+                                               VisibilityKind::Protected));
+  }
```

Comment:
`cirVisAttrStr` is not used and we can simply use `parseOptionalKeyword("hidden")` instead:

```cpp
if (parser.parseOptionalKeyword("hidden").succeeded()) {
  // ...
} else if (parser.parseOptionalKeyword("protected").succeeded()) {
  // ...
}
```